posts/2024-09-10-24.0.0.9.adoc

```diff
@@ -0,0 +1,311 @@
+
+The partitioned attribute configuration is opt-in and behaves much like the SameSite configuration. The `samesite` channel configuration applies to all cookies, while the `httpSession` and `webAppSecurity` configurations apply to their respective cookies. It's important to note that the `httpSession` and `webAppSecurity` configurations take precedence over the channel configuration. The default value for these two attributes is `defer`, which means they defer to the channel configuration. As for channel configuration, its default value is `false`, which means the `Partitioned` attribute is not added.
+
+Depending on which configuration you use to declare the `Partitioned` attribute, Liberty uses one of three attributes.
+
+The following example shows how to set the `cookiePartitioned` attribute for the HTTP session cookie on the `httpSession` attribute in your `server.xml` file:
+
+[source,xml]
+----
+<httpSession cookieSameSite="None" cookiePartitioned="defer|true|false"/>`
+----
+
+The following example shows how to set the `partitionedCookie` attribute for LTPA and JWT security cookies on the `webAppSecurity` attribute in your `server.xml` file:
+
+[source,xml]
+----
+<webAppSecurity sameSiteCookie="None" partitionedCookie="defer|true|false"/>`
+----
+
+The following example shows how to set the `partitioned` attribute for other cookies on the `httpEndpoint` attribute in your `server.xml` file:
+
+[source,xml]
+----
+<httpEndpoint id="defaultHttpEndpoint"
+              httpPort="9080"
+              httpsPort="9443" >
+   <samesite none="*" partitioned="true|false"/>
+</httpEndpoint>
+----
+
+
+Alternatively, you can set `Partitioned` by using the `Set-Cookie` header with the following two `HttpServletResponse` APIs:
+
+* link:https://openliberty.io/docs/latest/reference/javadoc/liberty-jakartaee10-javadoc.html?path=liberty-jakartaee10-javadoc/jakarta/servlet/http/HttpServletResponse.html[HttpServletResponse.setHeader]
+* link:https://openliberty.io/docs/latest/reference/javadoc/liberty-jakartaee10-javadoc.html?path=liberty-jakartaee10-javadoc/jakarta/servlet/http/HttpServletResponse.html[HttpServletResponse.addHeader]
+
+For more information, including a visual example, see link:https://github.com/privacycg/CHIPS?tab=readme-ov-file#chips-cookies-having-independent-partitioned-state[CHIPS (Cookies Having Independent Partitioned State)] on GitHub.
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/29571
+// Contact/Reviewer: jhanders34
+// // // // // // // //
+[#versionless]
+== Versionless features for Java / Jakarta EE Container Liberty features
+
+In 24.0.0.8, Open Liberty introduced versionless Java EE and Jakarta EE features.  Those new versionless features allow you to easily use features without needing to know what feature version to use.
```

Comment:
Didn't know if you wanted to add a reference link back to https://openliberty.io/blog/2024/08/13/24.0.0.8.html#versionless

Reply:
I linked to the versionless features doc at the end of this section instead